packages/loader/driver-utils/src/runWithRetry.ts

```diff
@@ -38,22 +42,58 @@ export interface IProgress {
     onRetry?(delayInMs: number, error: any): void;
 }
 
-export async function runWithRetry<T>(
+export interface IProgress2 {
+    /**
+     * Abort signal used to cancel operation
+     * Note that most of the layers do not use this signal yet. We need to change that over time.
+     * Please consult with API documentation / implementation.
+     * Note that  number of layers may not check this signal while holding this request in a queue,
+     * so it may take a while it takes effect. This can be improved in the future.
+     * Layers in question are:
+     *    - driver (RateLimiter)
+     *    - runWithRetry
+     */
+    cancel?: AbortSignal & { reason: string; };
```

Comment:
Ignore this where I'm tacking on reason. I haven't been able to look into why it's not there and just wanted it for prototyping